testing/steps/test_zip_data.py

```diff
@@ -0,0 +1,115 @@
+
+    # Then
+    assert data_container.data_inputs.shape == (SHAPE_3D[0], SHAPE_3D[1], SHAPE_3D[2] * 2)
+    assert data_container.expected_outputs.shape == (SHAPE_3D[0], SHAPE_3D[1], SHAPE_3D[2] * 2)
+    assert np.array_equal(data_container.data_inputs[..., -SHAPE_3D[2]:], data_container_3d_second.data_inputs)
+    assert np.array_equal(data_container.expected_outputs[..., -SHAPE_3D[2]:], data_container_3d_second.expected_outputs)
+
+
+def test_zip_data_should_merge_2d_with_3d():
+    # Given
+    data_inputs_3d, expected_outputs_3d = _create_data_source(SHAPE_3D)
+    data_inputs_2d, expected_outputs_2d = _create_data_source(SHAPE_2D)
+    data_container_2d = DataContainer(data_inputs=data_inputs_2d, expected_outputs=expected_outputs_2d)
+    data_container_3d = DataContainer(data_inputs=data_inputs_3d, expected_outputs=expected_outputs_3d) \
+        .add_sub_data_container('2d', data_container_2d)
+
+    # When
+    p = Pipeline([
+        ZipData(data_sources=['2d'])
+    ])
+
+    data_container_3d = p.handle_transform(data_container_3d, ExecutionContext())
+
+    # Then
+    assert data_container_3d.data_inputs.shape == (SHAPE_3D[0], SHAPE_3D[1], SHAPE_3D[2] + 1)
+    assert data_container_3d.expected_outputs.shape == (SHAPE_3D[0], SHAPE_3D[1], SHAPE_3D[2] + 1)
+    assert np.array_equal(data_container_3d.data_inputs[..., -1], data_container_2d.data_inputs)
+    assert np.array_equal(data_container_3d.expected_outputs[..., -1], data_container_2d.expected_outputs)
+
+
+def test_zip_data_should_merge_1d_with_3d():
+    # Given
+    data_inputs_3d, expected_outputs_3d = _create_data_source(SHAPE_3D)
+    data_inputs_1d, expected_outputs_1d = _create_data_source(SHAPE_1D)
+    data_container_1d = DataContainer(data_inputs=data_inputs_1d, expected_outputs=expected_outputs_1d)
+    data_container = DataContainer(data_inputs=data_inputs_3d, expected_outputs=expected_outputs_3d) \
+        .add_sub_data_container('1d', data_container_1d)
+
+    # When
+    p = Pipeline([
+        ZipData(data_sources=['1d'])
+    ])
+
+    data_container = p.handle_transform(data_container, ExecutionContext())
+
+    # Then
+    broadcasted_data_inputs_1d = np.broadcast_to(np.expand_dims(data_container_1d.data_inputs, axis=-1),
+                                                 shape=(SHAPE_3D[0], SHAPE_3D[1]))
+    broadcasted_expected_outputs_1d = np.broadcast_to(np.expand_dims(data_container_1d.expected_outputs, axis=-1),
+                                                      shape=(SHAPE_3D[0], SHAPE_3D[1]))
+
+    assert np.array_equal(data_container.data_inputs[..., -1], broadcasted_data_inputs_1d)
+    assert np.array_equal(data_container.expected_outputs[..., -1], broadcasted_expected_outputs_1d)
+
+    assert data_container.data_inputs.shape == (SHAPE_3D[0], SHAPE_3D[1], SHAPE_3D[2] + 1)
+    assert data_container.expected_outputs.shape == (SHAPE_3D[0], SHAPE_3D[1], SHAPE_3D[2] + 1)
```

Comment:
Also do a `BatchZipDataContainer` or something else: 
`# shape of data_container.data_inputs: (batch_size, ((time_steps_3D, features_3D), features_2D))`